lib/sla.js

```diff
@@ -34,62 +38,133 @@ function designateMixingBridge(client) {
 }
 
 /**
- * Adds created channel to the bridge.
- * @param {Object} channel - Created channel to be added to the bridge.
- * @param {Object} bridge - Bridge that channel is to be added to.
+ * Adds inbound and originated channels to the bridge.
+ * @param {Object} inbound - Incoming channel to be added to the bridge.
+ * @param {Object} dialed - Created/dialed channel to be added as well
+ * @param {Object} bridge - Bridge that channels are to be added to.
  * @return {Q} Q promise object.
  */
-function addChannelsToBridge(channel, bridge) {
+function addChannelsToBridge(inbound, dialed, bridge) {
   var defer = Q.defer();
-  console.log('Adding channel to bridge %s', bridge.id);
+  var numInBridge = 0;
+  console.log('Adding channels to bridge %s', bridge.id);
   var addChannel = Q.denodeify(bridge.addChannel.bind(bridge));
-  addChannel({channel: channel.id})
+  addChannel({channel: [inbound.id, dialed.id]})
     .catch(function (err) {
       defer.reject(err);
     });
-  bridge.once('ChannelEnteredBridge', function(event, bridge) {
-    console.log('Channel %s has entered the bridge', bridge.channel.id);
-  });
-  bridge.once('ChannelLeftBridge', function(event, bridge) {
-    console.log('Channel %s has left the bridge', bridge.channel.id);
-    defer.resolve('Application completed');
-  });
+  function channelEntered(event, object) {
+    console.log('Channel %s has entered the bridge', object.channel.id);
+    numInBridge += 1;
+  }
+  bridge.on('ChannelEnteredBridge', channelEntered);
+  function channelLeft(event, object) {
+    numInBridge -= 1;
+    console.log('Channel %s has left the bridge', object.channel.id);
+    if(numInBridge === 1) {
+      if(object.channel.id === inbound.id) {
+        console.log('Hanging up dialed channel %s', dialed.id);
+        var hangupDialed = Q.denodeify(dialed.hangup.bind(dialed));
+        hangupDialed()
+          .catch(function (err) {
+            defer.reject(err);
+          });
+      }
+      else if(object.channel.id === dialed.id) {
+        console.log('Hanging up inbound channel %s', inbound.id);
+        var hangupInbound = Q.denodeify(inbound.hangup.bind(inbound));
+        hangupInbound()
+          .catch(function (err) {
+            defer.reject(err);
```

Comment:
I would type this error so the caller can handle it without blowing up the program. It's good to know that a hangup failed but it should not blow up the entire program considering the channel has left at this point.